features/stop_on_failure.feature

```diff
@@ -1,212 +1,124 @@
-Feature: Stop on failure
+Feature: Stop on failure via config
   In order to stop further execution of steps when first step fails
   As a feature developer
-  I need to have a --stop-on-failure option
+  I need to have a stop-on-failure option on the CLI or in config
```

Comment:
I have combined the previous `stop_on_failure` and `stop_on_failure_config` features. There was a lot of overlap between them, and some things that weren't actually covered that well.

It feels like "stop on failure" is a single feature, and whether it's controlled by config or a CLI flag (and what happens if both are present) are details within that, so I think better we group the coverage together.